astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -1242,6 +1243,27 @@ def test_two_argument_ufunc_outer(self):
         assert np.all(s13_greater_outer == check13_greater_outer)
 
 
+@dataclasses.dataclass
+class DuckQuantity:
+    data: u.Quantity
+
+    @property
+    def unit(self) -> u.UnitBase:
+        return self.data.unit
+
+
+@pytest.mark.parametrize('ufunc', (np.add, np.multiply, np.less))
+@pytest.mark.parametrize('quantity', (1 * u.m, [1, 2] * u.m))
+@pytest.mark.parametrize('other', (object(), DuckQuantity(u.mm)))
+@pytest.mark.parametrize('out', (None, 'empty'))
```

Comment:
I like that you test `out` but it would be good to also test something like `np.negative(duck, out=quantity)` since those would end up in `__array_ufunc__` as well.

Reply:
@mhvk , I've reorganized the tests to try and accomplish this, let me know what you think of my attempt.